### PR TITLE
xml2obj: not every amdSec contains a techMD

### PR DESCRIPTION
**Pull request by mistydemeo**
Fixes #8431.

### CODE REVIEW

File: src/archivematicaCommon/lib/xml2obj.py [hunk 1]
```diff
@@ -13,6 +13,11 @@ def mets_file(src):
     
     file_uuid = None
     for amd in raw.mets_amdSec:
+        # Some amdSecs are other types of metadata (for instance sourceMD),
+        # and we don't care about those; skip them.
+        if amd.mets_techMD is None:
```
Comment (Hwesta) [inline src/archivematicaCommon/lib/xml2obj.py:18]: Is this specifically None?  `if not amd.mets_techMD`?

**Comment by Hwesta** [comment]
:+1:

File: src/archivematicaCommon/lib/xml2obj.py (hunk 1 above)
Comment (mistydemeo) [inline src/archivematicaCommon/lib/xml2obj.py:18]: It is, yes! `DataNode.__getattr__` returns `None` specifically if the requested attribute isn't present.

Comment (Hwesta) [inline src/archivematicaCommon/lib/xml2obj.py:18]: Then `is None` is probably better. :)

**Pull request merged**